ldap/servers/slapd/libglobs.c

```diff
@@ -9195,15 +9195,15 @@ config_set_value(
     case CONFIG_SPECIAL_REFERRALLIST:
         /* referral list is already an array of berval* */
         if (value)
-            slapi_entry_attr_replace(e, cgas->attr_name, (struct berval **)*value);
+            slapi_entry_attr_replace(e, cgas->attr_name, *(struct berval ***)value);
```

Comment:
We still have a warning here:

`stack-based buffer over-read [CWE-126] [-Wanalyzer-out-of-bounds]`

Reply:
Will something like this calm it?

``` c
    /* referral list is already an array of berval* */
    if (value) {
        struct berval **referral_list = *(struct berval ***)value;
        if (referral_list) {
            slapi_entry_attr_replace(e, cgas->attr_name, referral_list);
        } else {
            slapi_entry_attr_set_charptr(e, cgas->attr_name, "");
        }
    } else {
        slapi_entry_attr_set_charptr(e, cgas->attr_name, "");
    }
    break;
 ```

Reply:
FYI: I got 10 warnings for libglobs.c, I suspect it is again the false positive about following different branches in different  switches